docs/development-instructions.rst

```diff
@@ -1,30 +1,139 @@
 Development Instructions
-===========================
+=========================
 
-1. Clone the `ArduinoCore-API <https://github.com/arduino/ArduinoCore-API>`_ repo to any convenient location::
+..
+   TODOS:
+   - Contribution guidelines (move here from Contributing.md)
+   - Code conventions
+   - ....
+   - Tools installation:
+      - uncrustify
+      - python ?
+      - pip ?
+      - pre-commit hook?
+      - spellchecker
+      - ... 
+   - Installation the Arduino PSoC6 core
+   - Setting up the development environment
 
-   git clone https://github.com/arduino/ArduinoCore-API.git
+Environment setup
+------------------
 
-2. Locate in your computer where the Arduino15 packages are installed. 
-   e.g. Windows: c:/Users/%USERNAME%/AppData/local/Arduino15
-        Linux:   ~/.arduino15/
+.. note::
+   | The development environment setup is (currently) only supported on **Linux** |:penguin:|. 
+   | As many tools and scripts are not cross-platform, some of then will not directly work on Windows |:abcd:| or macOS |:apple:|.
 
-3. Change directories::
 
-   cd packages/Infineon-psoc/hardware/psoc/0.0.1
+0. Obviously |:neutral_face:|, install `Arduino IDE (2.0 or higher) <https://docs.arduino.cc/software/ide-v2/tutorials/getting-started/ide-v2-downloading-and-installing/>`_ or `Arduino CLI (1.0.0 or higher) <https://arduino.github.io/arduino-cli/0.24/installation/>`_.
```

Comment:
An empty line between 0 and 1 would look good. But not sure if this will appear in the readthedocs page. There is an empty line between 7 and 8 but is not showing in the page.

Reply:
I believe that is because of the comments I inserted there, but we will remove some of them, as we get all those things clear.